### ISSUE TITLE
All entities become state unknown when one or more entities are set to disabled

### ISSUE THREAD

**Issue by cbornemann** [jm-73/Indego#234]
After install some entities of the mower becomes state unknown, when i rename the id of the state sensor or when i add the new lawn-mower-control via the configuration checkbox.

Installed Version: 5.7.0
HA-Version: 2024.6.3

![Screenshot_20240621_193206_io homeassistant companion android](https://github.com/jm-73/Indego/assets/28005576/234f4133-7b29-42c3-8303-3022daba8e85)

Error-log says:

```
Logger: homeassistant
Quelle: helpers/entity.py:965
Erstmals aufgetreten: 19:01:55 (10 Vorkommnisse)
Zuletzt protokolliert: 19:15:27

Error doing job: Task exception was never retrieved (None)
Traceback (most recent call last):
  File "/config/custom_components/indego/__init__.py", line 495, in _initial_update
    self.set_online_state(False)
  File "/config/custom_components/indego/__init__.py", line 658, in set_online_state
    self.entities[ENTITY_MOWER_STATE_DETAIL].set_cloud_connection_state(online)
  File "/config/custom_components/indego/mixins.py", line 83, in set_cloud_connection_state
    self.async_schedule_update_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 1254, in async_schedule_update_ha_state
    self.async_write_ha_state()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 997, in async_write_ha_state
    self._async_verify_state_writable()
  File "/usr/src/homeassistant/homeassistant/helpers/entity.py", line 965, in _async_verify_state_writable
    raise RuntimeError(f"Attribute hass is None for {self}")
RuntimeError: Attribute hass is None for <entity unknown.unknown=unknown>

```

Reinstallation of Indego gives error:

```
Logger: homeassistant.util.package
Quelle: util/package.py:55
Erstmals aufgetreten: 19:46:02 (2 Vorkommnisse)
Zuletzt protokolliert: 19:46:06

Invalid requirement 'git+https://github.com/jm-73/pyIndego.git@3.2.0'
```

For the moment, i go back to 5.6.1. Here are all entities working perfectly.

**Comment by kimzeuner** [comment]
Not sure if it is the solution but maybe you can have at look at this issue from beta version [#226](https://github.com/jm-73/Indego/issues/226)

**Comment by cbornemann** [comment]
Not working for me, i have try it.

**Comment by sander1988** [comment]
The "invalid requirement" warning can be ignored. I have already checked it.

The runtime error causes the behavior you describe.

Can you explain what you mean with "changing the id"?

**Comment by cbornemann** [comment]
When i change the entity-id. But the problem occures randomly at one of the entities.

**Comment by sander1988** [comment]
The entity id of which entity? Please provide details, so I can try to do the exact same thing in my setup.

**Comment by cbornemann** [comment]
Now i have reinstalled the integration.

Then i have rename the entity-id´s, all works fine:

![1](https://github.com/jm-73/Indego/assets/28005576/2adf4889-6cf8-4ec9-8bd2-9fa80cebd58f)

But when i reload now the integration, all entities become these states:

![3](https://github.com/jm-73/Indego/assets/28005576/2a694f2e-ce9f-4959-a682-f42c3f2fa617)

I don´t know, before update of 5.7.0 all works fine.

**Comment by cbornemann** [comment]
Okay now i think i know the problem.

At this point, when i disable any entity and reload the integration, the integration crash.

But when i re-enable the disabled entities all works.

**Comment by cbornemann** [comment]
Okay the problem entities are: **mower_state_detail** and **runtime_total**.

When i disable one of these entities and reload the integration, the integration crashed.

**Comment by sander1988** [comment]
I guess disabling the entity is the problem here. Can you re-enable and restart HA?

**Comment by sander1988** [comment]
I have tested it. Caused by disabling one or more entities. It breaks updating the state update call.

**Workaround:** Do **not** disable entities; you can use the hide toggle for now.

**Comment by cbornemann** [comment]
Thank you, thats it!

**Comment by sander1988** [comment]
Thank you for confirming! We try to fix this in the next release. But workaround should be ok for now.

**Comment by sander1988** [comment]
Please test with pre-release 5.7.1 to see if the issue has been resolved.

**Comment by cbornemann** [comment]
Problem is fixed with 5.7.2! Thank you!

**Comment by WhyLev** [comment]
🤖 **AI Summary of the solution:**

The issue with entities becoming unknown when some are disabled was fixed in version 5.7.2. The workaround was to not disable entities.
